zml/aio/torch.zig

```diff
@@ -312,102 +312,97 @@ pub const PickleData = struct {
                 try self.parseValue(allocator, store, prefix, build.args);
             },
             .pers_id => |pers_id| try self.parseValue(allocator, store, prefix, pers_id.ref),
-            .seq => |*seq| switch (seq[0]) {
-                .list, .tuple, .set, .frozen_set => {
-                    const elemCheck = struct {
-                        fn call(comptime T: ValueType) fn (v: Value) bool {
-                            return struct {
-                                fn call(val: Value) bool {
-                                    return val == T;
+            .seq => |seq| {
+                const seq_type, const seq_values = seq;
+                switch (seq_type) {
+                    .list, .tuple, .set, .frozen_set => {
+                        if (seq_values.len == 0) return;
+                        var valid_slice = true;
+                        switch (seq_values[0]) {
```

Comment:
I think this would gain in clarity if you first checked that seq is a valid slice, then do the parsing.

Reply:
make `isValidSlice` into an helper:

```
fn isValidSlice(seq: Seq) ValueType {
  if (seq.values.len == 0) return true;   
  const t: ValueType = seq.values[0];
  switch (t) {
    .int64, float64, bool => {},
    else => return false,
  }
  for (seq.values) |v| {
    if (v != t) return false;
   }
  return true;
}
```

Reply:
Yea, that was the previous approach; I can definitely revert, but I do think this is leads to a performance hit. 

Under the previous approach, assuming all values in the slice are of the same type (i.e. the slice is valid), we would iterate over each element twice (once to validate the type and a second time to parse/store the value).

Under this new approach, if all values are of the same type, they are only iterated over once.

Under this new approach, if elements are not all of the same type, there is no performance difference as we end up iterating twice over the uniform elements that precede the element(s) that have a different type, any elements following the non-matching type are only iterated over once. In this case, there is no difference from the previous approach.

IMO, we keep the updated approach as most metadata is likely to be an array of the same type of values, especially large arrays, which is where the old approach would have the most impact on performance.

Reply:
fair point, let's optimize for the more frequent use case.